lightblue-migration-monitor/src/main/java/com/redhat/lightblue/migrator/monitor/Monitor.java

```diff
@@ -0,0 +1,93 @@
+import com.redhat.lightblue.client.LightblueException;
+import com.redhat.lightblue.client.Literal;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.response.LightblueDataResponse;
+import com.redhat.lightblue.migrator.MigrationConfiguration;
+import com.redhat.lightblue.migrator.MigrationJob;
+
+public class Monitor {
+
+    private final LightblueClient lightblueClient;
+
+    public Monitor(MonitorConfiguration cfg) {
+        if (cfg.getClientConfig() != null) {
+            lightblueClient = new LightblueHttpClient(cfg.getClientConfig());
+        } else {
+            lightblueClient = new LightblueHttpClient();
+        }
+    }
+
+    public void runCheck(final Notifier... notifiers) throws LightblueException {
+        List<String> configurationsMissingJobs = new ArrayList<>();
+        for (MigrationConfiguration cfg : findMigrationConfigurations()) {
+            long period = parsePeriod(cfg.getPeriod());
+            Date endDate = new Date();
+            Date startDate = new Date(endDate.getTime() - period);
+
+            int jobs = countMigrationJobs(cfg.getConfigurationName(), startDate, endDate);
+            if (jobs <= 0) {
+                configurationsMissingJobs.add(cfg.getConfigurationName());
+            }
+        }
+
+        if (configurationsMissingJobs.isEmpty()) {
+            for (Notifier n : notifiers) {
+                n.sendSuccess();
+            }
+        } else {
+            for (Notifier n : notifiers) {
+                n.sendFailure(configurationsMissingJobs);
+            }
+        }
+    }
+
+    /**
+     * Returns the period in msecs
+     */
+    private static long parsePeriod(String periodStr) {
+        PeriodFormatter fmt = PeriodFormat.getDefault();
+        Period p = fmt.parsePeriod(periodStr);
+        return p.toStandardDuration().getMillis();
+    }
+
+    private MigrationConfiguration[] findMigrationConfigurations() throws LightblueException {
+        DataFindRequest findConfigurations = new DataFindRequest(MigrationConfiguration.ENTITY_NAME);
+        findConfigurations.where(Query.withValue("period", Query.neq,
+                Literal.pojo(null)));
+        findConfigurations.select(
+                Projection.includeField("configurationName"),
+                Projection.includeField("period"));
+
+        return lightblueClient.data(findConfigurations, MigrationConfiguration[].class);
+    }
+
+    private int countMigrationJobs(String configurationName, Date startDate, Date endDate) throws LightblueException {
+        DataFindRequest findJobs = new DataFindRequest(MigrationJob.ENTITY_NAME);
+        findJobs.where(
+            Query.and(
+                Query.withValue("configurationName", Query.eq, configurationName),
```

Comment:
We can use scheduled date as an indication that the consistency checker is
still alive. There should be new jobs for every "period" with scheduled
date close to what the period should be.

On Tue, May 24, 2016 at 3:16 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> lightblue-migration-monitor/src/main/java/com/redhat/lightblue/migrator/monitor/Monitor.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/356#discussion_r64471153>
> :
>
> > +    private MigrationConfiguration[] findMigrationConfigurations() throws LightblueException {
> > +        DataFindRequest findConfigurations = new DataFindRequest(MigrationConfiguration.ENTITY_NAME);
> > +        findConfigurations.where(Query.withValue("period", Query.neq,
> > +                Literal.pojo(null)));
> > +        findConfigurations.select(
> > +                Projection.includeField("configurationName"),
> > +                Projection.includeField("period"));
> > +
> > +        return lightblueClient.data(findConfigurations, MigrationConfiguration[].class);
> > +    }
> > +
> > +    private int countMigrationJobs(String configurationName, Date startDate, Date endDate) throws LightblueException {
> > +        DataFindRequest findJobs = new DataFindRequest(MigrationJob.ENTITY_NAME);
> > +        findJobs.where(
> > +            Query.and(
> > +                Query.withValue("configurationName", Query.eq, configurationName),
>
> The point is to monitor for the creation of jobs, not that there's a
> contiguous range of jobs being created. That may be a separate issue to
> tackle. Could we use the scheduled date to just check that we're getting at
> least something generated for each configured and active migration?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/356/files/d9ae8a0dbee0850dc48d9d901a915525c4540c70#r64471153>
>